### DISCUSSION TITLE
How do you quickly search in Flow Launcher within the current folder you are in File Explorer ?

### DISCUSSION THREAD

**Discussion by oeloo** [Flow-Launcher/Flow.Launcher#2001] [Q&A]
Hi, I really like Flow Launcher search capabilities, but I am annoyed by one thing : I really need to be able to search only in the current folder (and sub-folders) I am currently watching in File Explorer window. With Windows search engine, I just had to type in the search box on the top-right :
![image](https://user-images.githubusercontent.com/98819304/227135769-2a6406f7-71c7-4c9f-819c-13c148e129ca.png)
Is there a way to search from the folder you are currently watching in File Explorer with Flow Launcher ? Regards

**Comment by jjw24** [comment]
Couple of ways:

1. Go to that folder in flow and use '>' which will search everything in the folder and subfolders.
2. If you usually have the explorer window open, setup Custom Query Hotkey pointing to the custom built-in get current window variable. This way when you trigger the hotkey you will get the path of your currently opened explorer window and then follow the first option to search.
3. Just search the name of the file/folder in flow. If results not coming up as you expect them to (assuming you are using Windows Search engine) you may need to configure your indexed paths.

**Reply by oeloo** [reply]
> 1. Go to that folder in flow and use '>' which will search everything in the folder and subfolders.

Sorry I do not get it :  _`>`_ in Flow Launcher is a for a command prompt, so I do not understand the link with my File Explorer current folder.


> 2\. If you usually have the explorer window open, setup Custom Query Hotkey pointing to the custom built-in get current window variable. This way when you trigger the hotkey you will get the path of your currently opened explorer window and then follow the first option to search.

How can I configure this ? I have made no modification on these default settings :
![image](https://user-images.githubusercontent.com/98819304/227152726-04f3da30-b8b2-4edd-8217-ca5f4d50b42a.png)
What is the _`active_explorer_path`_ ?


> 3. Just search the name of the file/folder in flow. If results not coming up as you expect them to (assuming you are using Windows Search engine) you may need to configure your indexed paths.

It is properly indexed, I am using Everything as search engine :
![image](https://user-images.githubusercontent.com/98819304/227153437-e503c5af-beee-4ba7-900b-527b33655f0e.png)

**Reply by jjw24** [reply]
> > 1. Go to that folder in flow and use '>' which will search everything in the folder and subfolders.
> 
> Sorry I do not get it : _`>`_ in Flow Launcher is a for a command prompt, so I do not understand the link with my File Explorer current folder.

It is a hotkey for Shell plugin but is different when used as part of Explorer query:
![image](https://user-images.githubusercontent.com/26427004/227365296-78e94321-056c-440e-b073-685acf094686.png)

or just `c:\Program Files\>` to see everything in Program Files

**Reply by jjw24** [reply]
> > 2. If you usually have the explorer window open, setup Custom Query Hotkey pointing to the custom built-in get current window variable. This way when you trigger the hotkey you will get the path of your currently opened explorer window and then follow the first option to search.
> 
> How can I configure this ? I have made no modification on these default settings

Like this :)

![image](https://user-images.githubusercontent.com/26427004/227367162-ee151a94-563b-406b-be47-6cbc42d6396a.png)

**Reply by oeloo** [reply]
Thank you for the clarification about the `>` double usage and the info about the shortcut.

When I use the shortcut in C:\Windows for example, I get this :
![image](https://user-images.githubusercontent.com/98819304/227450487-19110809-151b-449d-9109-bb15e4a5b665.png)

but to actually search inside I have to add `>.dll` :
![image](https://user-images.githubusercontent.com/98819304/227451256-df59e3a6-ca9e-4cbc-a5db-5f1ed0392bd7.png)

Is there a way to simplify this and just need to type `.dll` : the `>` should be already be there in my opinion, why requesting the user to type it each time ?

In Voidtools Everything, when you use its shortcut from File Explorer you are directly prompted with the current folder and a space: so you just have to type `.dll`
![image](https://user-images.githubusercontent.com/98819304/227451003-fbfee63d-8334-4d5a-a06b-882ae064d051.png)
